### PR TITLE
DOC gradio integration

### PR DESCRIPTION
**Pull request by merveenoyan**
This PR includes documentation for gradio integration. 
I added it to `plot_hf_hub.py` given it's hub related. However, for some reason it can't see there's an example data. 

```
ValueError: No example data found in README.md of skops-ci/hf_hub_example-96069c5a-d82d-4ffd-b1b8-5ac554c7f7ce - Cannot build gradio demo. See the README.md here: https://huggingface.co/scikit-learn/tabular-playground/blob/main/README.md for a reference on how to provide example data to your model.
```
Also I see [in the example](https://huggingface.co/skops-ci/hf_hub_example-96069c5a-d82d-4ffd-b1b8-5ac554c7f7ce/tree/main) there is example input. (the latest model I pushed and the one I got the above error from)
(I also tried on my local, it can build the interface on this Space with model pushed through this tutorial, see
[the model](https://huggingface.co/merve/hf_hub_example-87b3564e-aa7a-4409-990f-664cf8f11b1b/tree/main) and [the space](https://huggingface.co/spaces/merve/skops-test-gr))

pinging @freddyaboulton

### CODE REVIEW

File: examples/plot_hf_hub.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 import sklearn
 from huggingface_hub import HfApi
+import gradio as gr
```
Comment (adrinjalali) [inline examples/plot_hf_hub.py:24]: I think we should have a separate example for this.

Also need to add the dependency for doc build.

Note that this might actually break more often, since it might take a little while before gradio adapts to the latest skops. Adding this here means we expect gradio to work with the dev version of skops.

**Review by adrinjalali — COMMENTED** [review]
Thanks @merveenoyan

**Comment by merveenoyan** [comment]
@adrinjalali it looks like below: 
Ekran Resmi 2022-09-12 16 01 12

also now that we don't have it in examples, I removed it from min dependencies.

File: docs/hf_hub.rst [hunk 2]
```diff
@@ -111,3 +111,40 @@ returns the model output for a given input. It can be used as::
...
+prepended with "huggingface/" will load an interface for your model. The interface
+has a dataframe input that takes samples and a dataframe output to return 
+predictions. It also takes the example in the repository that is previously
+pushed with skops. Calling :func:`gr.Interface.launch()` will launch your application.
+
+.. code:: python
+
+    import gradio as gr
+    repo_id = "scikit-learn/tabular-playground"
+    gr.Interface.load(f"huggingface/{repo_id}").launch()
```
Comment (freddyaboulton) [inline docs/hf_hub.rst:134]: This works for me:

```python
gr.Interface.load("models/scikit-learn/tabular-playground").launch()
```

Comment (merveenoyan) [reply docs/hf_hub.rst:134]: same it worked for me as well, however, we have this script: https://skops.readthedocs.io/en/latest/auto_examples/plot_hf_hub.html#sphx-glr-auto-examples-plot-hf-hub-py in which we create an example model and repository automatically, and I previously put interface building at the end of it where gradio tried to retrieve that model and it did, but was failing to read the example input. I also checked that model specifically (that was pushed) and could build the interface myself. I feel like I'm missing something here 😅

TLDR; gradio can't create the UI to the model pushed through notebook when used inside the same notebook, but it can create UI when I call the model from somewhere else.

Comment (freddyaboulton) [reply docs/hf_hub.rst:134]: Gradio currently reads examples from README so that we can also load in models that were uploaded without skops. If you can read in the model yourself eventually, that makes me thing the examples are added to the README automatically after a delay?

Comment (merveenoyan) [reply docs/hf_hub.rst:134]: @freddyaboulton that might be the case (?)

File: skops/_min_dependencies.py [hunk 3]
```diff
@@ -27,6 +27,7 @@
     "sphinx-issues": ("1.2.0", "docs", None),
     "matplotlib": ("3.3", "docs", None),
     "pandas": ("1", "docs", None),
+    "gradio": ("3.3", "docs", None),
```
Comment (adrinjalali) [inline skops/_min_dependencies.py:30]: If we do a user guide instead of an example file, we don't need this dependency.

Comment (merveenoyan) [reply skops/_min_dependencies.py:30]: The tests were complaining that's why I had to add it back. I can remove again.

Comment (adrinjalali) [reply skops/_min_dependencies.py:30]: Do the tests actually run the code inside the `rst` files?  I don't remember.

**Comment by merveenoyan** [comment]
@adrinjalali mypy is complaining again 😂

**Comment by merveenoyan** [comment]
@adrinjalali can you review? can't ask for a re-review

File: docs/hf_hub.rst [hunk 4]
```diff
@@ -111,3 +111,40 @@ returns the model output for a given input. It can be used as::
...
+
+
+You can further customize your UI, add description, title, and more. If you'd 
+like to share your demo, you can set ``share`` to True in :func:`gr.Interface.launch()`.
+
+.. code:: python
+
+    title = "Supersoaker Defective Product Prediction"
+    description = "This model predicts Supersoaker production line failures. Drag and drop any slice from dataset or edit values as you wish in below dataframe component."
+    gr.Interface.load(f"huggingface/{repo_id}", title = title, description = description)
```
Comment (adrinjalali) [inline docs/hf_hub.rst:144]: lines too long

**Review by adrinjalali — COMMENTED** [review]
otherwise LGTM.

File: docs/hf_hub.rst [hunk 5]
```diff
@@ -111,3 +111,42 @@ returns the model output for a given input. It can be used as::
 
 In the above code snippet, ``res`` will be a :class:`numpy.ndarray` containing
 the model's output.
+
+.. _hf_hub_gradio:
+
+Easily build user interfaces to your scikit-learn models
+--------------------------------------------------------
+`gradio <https://gradio.app/>`__ is a python library that lets you create interfaces on your model.
+It has a class called :class:`Interface` that lets you create application
```
Comment (BenjaminBossan) [inline docs/hf_hub.rst:120]: The link to `Interface`, as well as below `launch`, is not properly generated. This is because we lack the intersphinx mapping to the gradio docs.

To add those, we need to add an entry [here](https://github.com/skops-dev/skops/blob/cfea7cb5a741a86365eaa9bd4d56a86cc46cbea5/docs/conf.py#L93) which should look similar to this:

```python
    "gradio": ("https://www.gradio.app/docs/", None),
```

However, when I tried, I got this error:

`intersphinx inventory 'https://www.gradio.app/docs/objects.inv' not fetchable due to <class 'requests.exceptions.HTTPError'>: 404 Client Error: Not Found for url: https://www.gradio.app/docs/objects.inv`

@freddyaboulton can you help?

File: docs/hf_hub.rst [hunk 6]
```diff
@@ -111,3 +111,42 @@ returns the model output for a given input. It can be used as::
...
+interfaces to your machine learning models.
+``gradio`` is integrated with skops, so you can load an interface with only one
+line of code. You need to initialize an interface, call load method with
```
Comment (BenjaminBossan) [inline docs/hf_hub.rst:123]: The sentence structure seems to be a bit weird to me:

> You need to initialize an interface, call load method with your repository identifier (...) prepended with "huggingface/" will load an interface for your model.

Not quite sure what it should actually be. Maybe split it into 2 sentences?

File: examples/plot_hf_hub.py [hunk 7]
```diff
@@ -162,6 +162,7 @@
 
 hub_utils.update_env(path=local_repo, requirements=["scikit-learn"])
 
+
```
Comment (BenjaminBossan) [inline examples/plot_hf_hub.py:165]: Why this change? We don't consistently use 2 empty lines.

**Review by BenjaminBossan — COMMENTED** [review]
I have a few more comments, the rest looks good to me.

File: examples/plot_hf_hub.py (hunk 7 above)
Comment (merveenoyan) [reply examples/plot_hf_hub.py]: I addressed this

File: docs/hf_hub.rst (hunk 5 above)
Comment (freddyaboulton) [reply docs/hf_hub.rst:120]: Looking into it @BenjaminBossan ! I think the problem is that we don't generate the file expected by sphinx. Can you link to the gradio website in a way that doesn't rely on intersphinx in the meantime?

**Comment by merveenoyan** [comment]
@BenjaminBossan I added a comment

File: docs/hf_hub.rst (hunk 5 above)
Comment (merveenoyan) [reply docs/hf_hub.rst:121]: I added a TODO comment in file in the meantime https://github.com/skops-dev/skops/pull/134/commits/b2a96a68b9a07ba2bcdc11016725e1d3c9324764

File: docs/hf_hub.rst [hunk 8]
```diff
@@ -111,3 +111,43 @@ returns the model output for a given input. It can be used as::
...
 In the above code snippet, ``res`` will be a :class:`numpy.ndarray` containing
 the model's output.
+
+.. _hf_hub_gradio:
+..
+   TODO: replace gradio link once gradio provides object.inv
+Easily build user interfaces to your scikit-learn models
+--------------------------------------------------------
+`gradio <https://gradio.app/>`__ is a python library that lets you create interfaces on your model.
+It has a class called :class:`Interface` that lets you create application
```
Comment (BenjaminBossan) [inline docs/hf_hub.rst:121]: `:class:Interface` and `:func:gr.Interface.launch()` still don't work (sorry for wonky quote but GH won't let me put backticks inside backticks D-:). For the users, it would be helpful to replace them for the time being with links to https://gradio.app/docs/#interface and https://gradio.app/docs/#launch

Comment (merveenoyan) [reply docs/hf_hub.rst]: fixed them

File: docs/hf_hub.rst [hunk 9]
```diff
@@ -111,3 +111,43 @@ returns the model output for a given input. It can be used as::
...
 In the above code snippet, ``res`` will be a :class:`numpy.ndarray` containing
 the model's output.
+
+.. _hf_hub_gradio:
+..
+   TODO: replace gradio link once gradio provides object.inv
+Easily build user interfaces to your scikit-learn models
+--------------------------------------------------------
+`gradio <https://gradio.app/>`__ is a python library that lets you create interfaces on your model.
+It has a class called `Interface https://gradio.app/docs/#interface`__ that lets you create application
```
Comment (BenjaminBossan) [inline docs/hf_hub.rst:121]:
```suggestion
It has a class called `Interface <https://gradio.app/docs/#interface>`__ that lets you create application
```
